TLM/TLM/UI/SubTools/SpeedLimits/SpeedLimitsTool.cs

```diff
@@ -407,15 +407,17 @@ public SpeedLimitsTool(TrafficManagerTool mainTool)
 
         /// <summary>Called by IObservable when observed event is fired (UI language change).</summary>
         public void OnUpdate(ModUI.EventPublishers.LanguageChangeNotification subject) {
-            // All text labels on the window are changing, better to hide and rebuild the window
+            // All text labels on the window are changing
+            // Close window, deactivate tool
             this.DestroyWindow();
             MainTool.SetToolMode(ToolMode.None);
         }
 
         /// <summary>Called by IObservable when observed event is fired (display MPH change).</summary>
         public void OnUpdate(ModUI.EventPublishers.DisplayMphNotification subject) {
-            // Does not close window
-            this.RecreateToolWindow();
+            // Close window, deactivate tool
+            this.DestroyWindow();
+            MainTool.SetToolMode(ToolMode.None);
```

Comment:
This change leads to the issue that speed limit palette is visible but Speed limit is inactive

Reply:
Made smarter window reset, it will destroy window if the tool wasn't active, or it will rebuild the window if the tool was active.